packages/xchain-thorchain-amm/src/wallet.ts

```diff
@@ -374,6 +416,23 @@ export class Wallet {
       const evmHelper = new EvmHelper(assetClient, this.thorchainQuery.thorchainCache)
       const hash = await evmHelper.sendDeposit(addParams)
       return { hash, url: assetClient.getExplorerTxUrl(hash) }
+    } else if (this.isUTXOChain(params.amount.asset)) {
+      const feeRates = await (assetClient as UTXOClient).getFeeRates(Protocol.THORCHAIN)
+      const addParams = {
+        wallIndex: 0,
+        asset: params.amount.asset,
+        amount: params.amount.baseAmount,
+        recipient: params.toAddress,
+        memo: params.memo,
+        feeRate: feeRates.fast,
+      }
+      try {
+        const hash = await assetClient.transfer(addParams)
+        return { hash, url: assetClient.getExplorerTxUrl(hash) }
+      } catch (err) {
+        const hash = JSON.stringify(err)
+        return { hash, url: assetClient.getExplorerAddressUrl(await assetClient.getAddressAsync()) }
+      }
```

Comment:
Is there any way to standardize this for all chains? I think this goes against the right way, the trend should be to eliminate these conditionals. In the case of EVMs they are 'indispensable' because the operation of TC demands it, but in this case the conditional is only motivated by the architecture of Xchain. We could standardize it for all strings as much as possible ? if not in this PR at least leave an issue created to eliminate this conditional. Over time this code is difficult to maintain.

Reply:
I agree with you. Let's find out a way to standardise this code, but I think it is better to do it in a different issue, this pull request is quite big